src/components/Ledger/LedgerView/LedgerFunds.js

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { withRouter } from 'react-router-dom';
+import { get } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+import { LIMIT_MAX } from '@folio/stripes-acq-components';
+
+import ConnectionListing from '../../ConnectionListing';
+import { groupFundFiscalYears } from '../../../common/resources';
+
+
+const LedgerFunds = ({ history, funds, currency, mutator, fiscalYears, resources }) => {
+  const openFund = useCallback(
+    (e, fund) => {
+      const path = `/finance/fund/view/${fund.id}`;
+
+      history.push(path);
+    },
+    [history],
+  );
+
+  const buildQuery = useMemo(() => {
+    const fundIds = funds.map(fund => `fundId="${fund.id}"`);
+    const fiscalYearsIds = fiscalYears.map(fiscalYear => `fiscalYearId="${fiscalYear.id}"`);
+
+    if (fundIds.length && fiscalYears.length) {
+      return `query=((${fundIds.join(' or ')}) AND (${fiscalYearsIds.join(' or ')}))`;
```

Comment:
imagine that there are 40 funds assigned to ledger. this request will probably fail. You can try it actually. 
I suggest to filter only by current fiscal year. I'm not sure why it's an array, but in any case it should be one or two values. 
Main list comes from request to `/funds`, here is only numbers, so you can extend the list items with grouped numbers by fundId.